### DISCUSSION TITLE
"No matching manifest for windows/amd64 x.x.x.x in the manifest list entries"

### DISCUSSION THREAD

**Discussion by whgibbo** [rancher-sandbox/rancher-desktop#2782] [General]
Hi,
Just installed Rancher Destkop 1.5.1 on Windows and when trying to pull an image in **powershell**

`docker pull docker/getting-started`

I get the following error:
```
Using default tag: latest
latest: Pulling from docker/getting-started
no matching manifest for windows/amd64 x,x,x,x in the manifest list entries
```
where x.x.x.x is the windows version.

Same issues using window **cmd** prompt.
When using doing this from the rancher-desktop wsl instance, there are no issues.. Image is pulled.

Was wondering if anybody had come across this and found a solution?

**Comment by mook-as** [comment]
That's _really_ odd, since we don't actually support running Windows containers in the first place (you'd have needed to pull `linux/amd64` instead).

Do you have Docker Desktop installed (or previously installed) and set to run Windows containers (which implies Windows Pro or higher)? If yes, please 1) identify what in `~/.docker/config.json` sets that (so we can change it for other people too), and 2) set it to use Linux containers.

Thanks!

**Reply by whgibbo** [reply]
> That's _really_ odd, since we don't actually support running Windows containers in the first place (you'd have needed to pull `linux/amd64` instead).
> 
> Do you have Docker Desktop installed (or previously installed) and set to run Windows containers (which implies Windows Pro or higher)? If yes, please 1) identify what in `~/.docker/config.json` sets that (so we can change it for other people too), and 2) set it to use Linux containers.
> 
> Thanks!

Hi,
I think I may have had Docker Desktop previously, but uninstalled it..  But never for Windows containers

Currently .docker/config.json is as follows:

`
{
	"auths": {
		"https://index.docker.io/v1/": {}
	},
	"credsStore": "wincred"
}`

**Reply by whgibbo** [reply]
@mook-as   What is the command to set to linux containers ?

**Reply by mook-as** [reply]
You can do `docker run --platform=linux/amd64` etc., but normally it doesn't need to set explicitly.
It looks like you can also set a `DOCKER_DEFAULT_PLATFORM` environment variable; perhaps you have that set to `windows/amd64`?

**Reply by whgibbo** [reply]
@mook-as   using:

`docker pull --platform=linux/amd64 docker/getting-started`
or
`docker pull --platform=linux docker/getting-started`


Both return **unsupported os linux**

I thought the --platform only worked when windows containers was enabled.. Could be wrong.

**Reply by mook-as** [reply]
Odd. `--platform` is only _useful_ when Windows containers are enabled (because otherwise the only platform you can choose is `linux/amd64`); but it looks like something is really confused such that it thinks Linux is unsupported. Which probably means you're managing to talk to some other docker backend, and not ours (we only ship the Linux one).

It's unclear how, though, since you don't have Docker Desktop anymore…

**Comment by jandubois** [comment]
Can you run `docker info` and show the output?

Also, could you switch to the `containerd` engine and see what happens when you `nerdctl pull` and image?

**Comment by whgibbo** [comment]
@jandubois 

docker info output using **dockerd(moby)**:

```Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Docker Buildx (Docker Inc., v0.8.2-docker)
  compose: Docker Compose (Docker Inc., v2.2.3)
Server:
 Containers: 7
  Running: 0
  Paused: 0
  Stopped: 7
 Images: 1
 Server Version: 20.10.17
 Storage Driver: windowsfilter
  Windows:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local logentries splunk syslog
 Swarm: inactive
 Default Isolation: hyperv
 Kernel Version: 10.0 19044 (19041.1.amd64fre.vb_release.191206-1406)
 Operating System: Windows 10 Enterprise Version 2009 (OS Build 19044.1889)
 OSType: windows
 Architecture: x86_64
 CPUs: 16
 Total Memory: 63.73GiB
 Name: PC-4WYY5M3
 ID: JT5X:7CQ3:RGA5:V5ZO:75MR:7266:7V4E:ZOY3:2C62:3L65:OVCC:Q2BZ
 Docker Root Dir: C:\ProgramData\docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

**Reply by jandubois** [reply]
This shows that you are running Docker Community Edition for Windows, not Rancher Desktop. You will have to uninstall/disable it to be able to use RD.

**Comment by whgibbo** [comment]
@jandubois   thanks I think was playing with Docker CE edition prior to Rancher Desktop..  Once I removed and reinstalled.. Issue resolved..
